third_party/libgcc/arm/lib1funcs.S

```diff
@@ -37,8 +37,9 @@ see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
 #define __prefer_thumb__
 #define @TODEFINE@
 #ifdef L_divsi3
-# define L_dvmd_tls /* for __aeabi_idiv0 */
+# define L_dvmd_tls       /* for __aeabi_idiv0    */
 # define L_aeabi_uldivmod /* for __aeabi_uldivmod */
+# define L_aeabi_ldivmod  /* for __aeabi_ldivmod  */
```

Comment:
I'm wondering why this is done this way rather than adding `add_libgcc_routines(aeabi_ldivmod)` to core/CMakeLists.txt, which then leaves the gcc file closer to the original?  I know you're just mirroring the `aeabi_uldivmod` so I'm asking about that too.  The comment here and the one in core/CMakeLists.txt talk about compling separately for each function, so I'm wondering why `divsi3` is special and happens to get bundled with some other routines.  Is it just that there happen to be no duplicate symbols (like `.Loop1`) when this set is compiled at once, so we save time?  Or do they need to be compiled at once for some reason?

Reply:
There are three lines in the ifdef:

```
# define L_dvmd_tls       /* for __aeabi_idiv0    */
# define L_aeabi_uldivmod /* for __aeabi_uldivmod */
# define L_aeabi_ldivmod  /* for __aeabi_ldivmod  */
```

Why is the first one different from the others? Shouldn't the three of them be moved to `core/CMakeLists.txt` or none of them? So:

``` bash
add_libgcc_routines(aeabi_idiv0)
add_libgcc_routines(aeabi_uldivmod)
add_libgcc_routines(aeabi_ldivmod)
```

I'm asking from absolute ignorance.

Reply:
Yeah, we're asking the same question.  Unfortunately the history shows that I was the one who wrote this :( and I cannot find any further notes.  I would have to guess that I just tried to compile as many at once as I could to save time, and split the ones with conflicting symbols (using the same label names, etc.).  Looking at it now it seems cleaner to do them all separately which reduces changes to the third-party file.  WDYT?

Reply:
I tries with https://github.com/umarcor/dynamorio/commit/9e06e105d1ead2f0327b52e70f30fb1c0f30993a. Also with:

```
add_libgcc_routines(idiv0)
add_libgcc_routines(uldivmod)
add_libgcc_routines(ldivmod)
```

Unfortunately, none of the work. Both fail with `undefined symbol: GLOBAL DEFAULT UND __aeabi_uldivmod`.